### PR TITLE
#169512289 fix date format in update requests

### PR DESCRIPTION
**Pull request by swaibat**
[Finishes #169512289]

#### What does this PR do?
refactor update requests
#### Description of Task to be completed?
- fix date format
- refactor reason field
- fix spelling errors

#### How should this be manually tested?
- clone the repository
- npm run test
- access update request route
- to see the new date format

#### Any background context you want to provide?
n/a
#### What are the relevant pivotal tracker stories?
[#61 169512289](https://www.pivotaltracker.com/story/show/169512289)
#### Questions:

### CODE REVIEW

File: src/validation/trips/index.js [hunk 1]
```diff
@@ -110,52 +110,42 @@ export const validateTrips = async (req, res, next) => {
...
-    }
-
-    if (!names.find(place => place === city.toLocaleLowerCase())) {
-      return resSend(res, 400, `we dint find ${city} in  ${destination.name}`);
-    }
-    next();
-  } catch (err) {
-    return resSend(res, 400, err.message);
+  const { country, city, departureTime } = req.body.trip;
+  if (departureTime && !departureTime.match(/[12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01])/)) {
```
Comment (elchroy) [inline src/validation/trips/index.js]: Can you put this inside a validator, if there is no one already to handle date format? In that way, all references to date formats will use the validator.

File: src/validation/trips/index.js [hunk 2]
```diff
@@ -110,52 +110,39 @@ export const validateTrips = async (req, res, next) => {
...
-  } catch (err) {
-    return resSend(res, 400, err.message);
+  const { country, city } = req.body.trip;
+  const destination = locations.getCountryByShort(country.toUpperCase());
+  const { suggestions, error, names } = helper.getCityData(destination.states, city);
+  if (error) {
+    return resSend(res, 400, error, { suggestions });
   }
+  if (!names.find(place => place === city.toLocaleLowerCase())) {
+    return resSend(res, 400, `we didnt find ${city} in  ${destination.name}`);
```
Comment (elchroy) [inline src/validation/trips/index.js]: Change this to `cannot find ${city} in ${destination.name}`

File: src/validation/trips/index.js [hunk 3]
```diff
@@ -110,52 +110,39 @@ export const validateTrips = async (req, res, next) => {
...
-      throw new Error('trips to the same city not allowed');
-    }
+  const { country, city, trip, returnTime } = req.body;
+  const origin = locations.getCountryByShort(country.toUpperCase());
+  const { suggestions, error, names } = helper.getCityData(origin.states, city);
+  if (error) {
+    return resSend(res, 400, error, { suggestions });
+  }
+  if (!names.find(place => place === city.toLocaleLowerCase())) {
+    return resSend(res, 400, `we dint find ${city} in  ${origin.name}`);
```
Comment (elchroy) [inline src/validation/trips/index.js]: Change this to `cannot find ${city} in ${origin.name}`

**Review by lemoissonM — COMMENTED** [review]
howesome work, I've left some comments

**Review by lemoissonM — DISMISSED** [review]
great work, I'm testing

**Review by Crispy-rw — APPROVED** [review]

**Review by lemoissonM — APPROVED** [review]

**Pull request merged**